clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemFactory.java

```diff
@@ -231,19 +255,10 @@ public SystemProducer getProducer(
       localVeniceZKHosts = legacyLocalVeniceZKHosts;
     }
 
-    String localControllerD2Service = config.get(VENICE_CHILD_CONTROLLER_D2_SERVICE);
-    if (isEmpty(localControllerD2Service)) {
-      LOGGER.info(
-          VENICE_CHILD_CONTROLLER_D2_SERVICE + " is not defined. Using " + LEGACY_VENICE_CHILD_CONTROLLER_D2_SERVICE);
-      localControllerD2Service = LEGACY_VENICE_CHILD_CONTROLLER_D2_SERVICE;
-    }
-
-    String parentControllerD2Service = config.get(VENICE_PARENT_CONTROLLER_D2_SERVICE);
-    if (isEmpty(parentControllerD2Service)) {
-      LOGGER.info(
-          VENICE_PARENT_CONTROLLER_D2_SERVICE + " is not defined. Using " + LEGACY_VENICE_PARENT_CONTROLLER_D2_SERVICE);
-      parentControllerD2Service = LEGACY_VENICE_PARENT_CONTROLLER_D2_SERVICE;
-    }
+    String localControllerD2Service =
+        config.get(VENICE_CHILD_CONTROLLER_D2_SERVICE, LEGACY_VENICE_CHILD_CONTROLLER_D2_SERVICE);
```

Comment:
BTW, it's not exactly the same logic as before; previously, the legacy config would also be used if the config exists and the config value is empty.

Reply:
That's okay. The `VENICE_CHILD_CONTROLLER_D2_SERVICE` config was added in a recent commit

Reply:
I just checked the prev commit and realized you're right. Will revert this change